Table/Table.cc

```diff
@@ -216,28 +281,29 @@ bool Table::ConstructFromFITS(bool header_only) {
     for (auto i = 1; i <= num_cols; i++) {
         auto& column = _columns.emplace_back(Column::FromFitsPtr(file_ptr, i, col_offset));
         // Resize column's entries vector to contain all rows
-        column->Resize(_num_rows);
+        column->Resize(allocated_rows);
         // Add columns to map
         if (!column->name.empty()) {
             _column_name_map[column->name] = column.get();
         }
     }
-    if (_num_rows) {
+    if (allocated_rows) {
         // Read entire table into a memory buffer
-        std::size_t size_bytes = total_width * _num_rows;
+        std::size_t size_bytes = total_width * allocated_rows;
         auto buffer = make_unique<uint8_t[]>(size_bytes);
         fits_read_tblbytes(file_ptr, 1, 1, size_bytes, buffer.get(), &status);
         // File is no longer needed after table is read
         fits_close_file(file_ptr, &status);
 
         // Dynamic schedule of OpenMP division, as some columns will be easier to parse than others
-#pragma omp parallel for default(none) schedule(dynamic) shared(num_cols, buffer, _num_rows, total_width)
+#pragma omp parallel for default(none) schedule(dynamic) shared(num_cols, buffer, allocated_rows, total_width)
         for (auto i = 0; i < num_cols; i++) {
-            _columns[i]->FillFromBuffer(buffer.get(), _num_rows, total_width);
+            _columns[i]->FillFromBuffer(buffer.get(), allocated_rows, total_width);
         }
     } else {
         fits_close_file(file_ptr, &status);
     }
+    _num_rows = rows;
```

Comment:
Is this intentionally set to `rows` and not `allocated_rows`?

Reply:
yes. `allocated_rows` is the number of rows to allocate for reading. When reading the header, this is zero. I'll add a comment

Reply:
actually, this is misleading. We need a new field to indicate the total rows available, not the total rows read into the table